weldx/transformations/local_cs.py

```diff
@@ -687,18 +692,36 @@ def interp_time(
                 "allowed. Also check that the reference time has the correct type."
             )
 
-        orientation = ut.xr_interp_orientation_in_time(self.orientation, time)
+        # calculate orientations
+        if self.orientation.ndim == 2:
+            orientation = self.orientation
+        elif time.max() <= self.time.min():
+            orientation = self.orientation.values[0]
+        elif time.min() >= self.time.max():
+            orientation = self.orientation.values[-1]
+        else:
+            orientation = ut.xr_interp_orientation_in_time(self.orientation, time)
```

Comment:
```suggestion
        # consider interpolation time with regard of LCS time for interpolation
        if self.orientation.ndim == 2:  # don't interpolate static
            orientation = self.orientation
        elif time.max() <= self.time.min():  # only use edge timestamp
            orientation = self.orientation.values[0]
        elif time.min() >= self.time.max():  # only use edge timestamp
            orientation = self.orientation.values[-1]
        else:  # full interpolation with overlapping times
            orientation = ut.xr_interp_orientation_in_time(self.orientation, time)
```

what do you think @vhirtham ?
Comments like this sometimes help me get back into the code

Reply:
Added the comments.  I gave the coordinate and orientation interpolations their own private methods to get the code a bit less crowded